txpipe/auxiliary_maps.py

```diff
@@ -0,0 +1,200 @@
+from .maps import TXBaseMaps
+import numpy as np
+from .base_stage import PipelineStage
+from .mapping import Mapper, FlagMapper, BrightObjectMapper, DepthMapperDR1
+from .data_types import MapsFile, HDFFile, ShearCatalog
+from .utils import choose_pixelization
+from .utils.calibration_tools import read_shear_catalog_type
+
+
+class TXAuxiliaryMaps(TXBaseMaps):
+    name = "TXAuxiliaryMaps"
+    """
+    This class generates:
+        - depth maps
+        - psf maps
+        - bright object maps
+        - flag maps
+    """
+    inputs = [
+        ("photometry_catalog", HDFFile),  # for mags etc
+        ("shear_catalog", ShearCatalog),  # for psfs
+        ("shear_tomography_catalog", HDFFile),  # for per-bin psf maps
+        ("source_maps", MapsFile),  # we copy the pixel scheme from here
+    ]
+    outputs = [
+        ("aux_maps", MapsFile),
+    ]
+
+    config_options = {
+        "chunk_rows": 100_000,
+        "sparse": True,
+        "flag_exponent_max": 8,  # flag bits go up to 2**8 by default
+        "psf_prefix": "psf_",  # prefix name for columns
+        "bright_obj_threshold": 22.0,  # The magnitude threshold for a object to be counted as bright
+        "depth_band": "i",  # Make depth maps for this band
+        "snr_threshold": 10.0,  # The S/N value to generate maps for (e.g. 5 for 5-sigma depth)
+        "snr_delta": 1.0,  # The range threshold +/- delta is used for finding objects at the boundary
+    }
+    # instead of reading from config we match the basic maps
```

Comment:
ah ok I see now that you're matching the basic -- so there's no scenarios where we just want to run this without the basic maps?

Reply:
Thanks, good point - I've added a standalone variant of the aux mapper that does this.
I think it's still useful to get the pixel scheme from the source maps in the default case, just to avoid risking mismatching resolutions, but the option is there now.